redfish-core/lib/managers.hpp

```diff
@@ -1922,6 +1922,70 @@ inline void setDateTime(std::shared_ptr<bmcweb::AsyncResp> aResp,
     }
 }
 
+/**
+ * @brief Retrieves bmc state properties over dbus
+ *
+ * @param[in] aResp     Shared pointer for completing asynchronous calls.
+ *
+ * @return None.
+ */
+inline void getBMCState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
+{
+    crow::connections::systemBus->async_method_call(
+        [aResp](const boost::system::error_code ec,
+                const std::variant<std::string>& bmcState) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error " << ec;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const std::string* state = std::get_if<std::string>(&bmcState);
+            if (!state)
+            {
+                messages::internalError(aResp->res);
+                return;
+            }
+            if (*state == "xyz.openbmc_project.State.BMC.BMCState.Ready")
+            {
+                aResp->res.jsonValue["PowerState"] = "On";
+                aResp->res.jsonValue["Status"]["State"] = "Enabled";
+                aResp->res.jsonValue["Status"]["Health"] = "OK";
+            }
+            else if (*state == "xyz.openbmc_project.State.BMC.BMCState."
+                               "Quiesced")
+            {
+                aResp->res.jsonValue["PowerState"] = "On";
+                aResp->res.jsonValue["Status"]["State"] = "Quiesced";
+                aResp->res.jsonValue["Status"]["Health"] = "Critical";
+            }
+            else if (*state == "xyz.openbmc_project.State.BMC.BMCState."
+                               "NotReady")
+            {
+                aResp->res.jsonValue["PowerState"] = "Starting";
```

Comment:
This is not a valid PowerState. PowerState shouldprobably be PoweringOn.
```
        "PowerState": {
            "enum": [
                "On",
                "Off",
                "PoweringOn",
                "PoweringOff",
                "Paused"
            ],
```            
            State needs to be Starting
            
 ```           
                    "State": {
            "enum": [
                "Enabled",
                "Disabled",
                "StandbyOffline",
                "StandbySpare",
                "InTest",
                "Starting",
                "Absent",
                "UnavailableOffline",
                "Deferring",
                "Quiesced",
                "Updating",
                "Qualified"
            ],
```

Reply:
done